### PR TITLE
add minquotas to podgroup spec

### PR DESCRIPTION
**Pull request by merryzhou**
Signed-off-by: merryzhou <merryzhou0882@gmail.com>

### CODE REVIEW

File: pkg/apis/scheduling/types.go [hunk 1]
```diff
@@ -183,6 +183,11 @@ type PodGroupSpec struct {
 	// if there's not enough resources to start all tasks, the scheduler
 	// will not start anyone.
 	MinResources *v1.ResourceList
+
+	// MinQuotas defines the minimal resource quota of members/tasks to run the pod group;
+	// if there's not enough resources to start all tasks, the scheduler
+	// will not start anyone.
+	MinQuotas *v1.ResourceList
```
Comment (Thor-wl) [inline pkg/apis/scheduling/types.go:190]: What's the difference between `MinResource` and `MinQuotas`? Is there a design doc for the feature?

Comment (merryzhou) [reply pkg/apis/scheduling/types.go:190]: https://github.com/volcano-sh/volcano/pull/1583

Comment (shinytang6) [reply pkg/apis/scheduling/types.go:190]: BTW, can we just use `MinResource` when considering resourceQuota? In other words, do we need to add a new `MinQuotas` field?

Comment (merryzhou) [reply pkg/apis/scheduling/types.go:190]: minResouce only contains request resources， but resourceQuota contains request & limits

Comment (Thor-wl) [reply pkg/apis/scheduling/types.go:190]: > minResouce only contains request resources， but resourceQuota contains request & limits

So if `resourceQuota` contains the function of `minResource`?

Comment (merryzhou) [reply pkg/apis/scheduling/types.go:190]: here is an example：
```
  minQuotas:
      count/pods: "1"
      cpu: "8"
      limits.cpu: "32"
      limits.ephemeral-storage: 50Gi
      limits.memory: 64Gi
      memory: 16Gi
      pods: "1"
      requests.cpu: "8"
      requests.memory: 16Gi
    minResources:
      cpu: "8"
      ephemeral-storage: 50Gi
      memory: 16Gi
```
so minQuotas contains the data of minResources。 but these two fields are for different scenarios。 and we cannot get rid of minresources.

Comment (shinytang6) [reply pkg/apis/scheduling/types.go:190]: got it

**Review by shinytang6 — CHANGES_REQUESTED** [review]
l think the `deepcopy` files should also be updated, maybe you should run `hack/update-codegen.sh`

**Comment by merryzhou** [comment]
> l think the `deepcopy` files should also be updated, maybe you should run `hack/update-codegen.sh`

Already ran。。

**Comment by shinytang6** [comment]
> > l think the `deepcopy` files should also be updated, maybe you should run `hack/update-codegen.sh`
> 
> Already ran。。

emmm, it may be generated into your gopath dir or somewhere else, it should be updated theoretically

**Comment by merryzhou** [comment]
> > > l think the `deepcopy` files should also be updated, maybe you should run `hack/update-codegen.sh`
> > 
> > 
> > Already ran。。
> 
> emmm, it may be generated into your gopath dir or somewhere else, it should be updated theoretically

done

File: pkg/apis/scheduling/types.go (hunk 1 above)
Comment (k82cn) [reply pkg/apis/scheduling/types.go:190]: sorry, I'm confused about this field. IMO, we should reuse minResource.

Comment (shinytang6) [reply pkg/apis/scheduling/types.go:190]: I think the main point here is that minResources is a value calculated based on limit and requests, but minQuotas can contain both limit and requests

Comment (merryzhou) [reply pkg/apis/scheduling/types.go:190]: > we should reuse minResource.

as @shinytang6 said, if we reuse minResource, the meaning of this field is changed. i don't think this is what we expected

Comment (k82cn) [reply pkg/apis/scheduling/types.go:190]: we need pay more attentation on similar new field :)

Comment (k82cn) [reply pkg/apis/scheduling/types.go:190]: is there anyway to enhance minResource?

Comment (merryzhou) [reply pkg/apis/scheduling/types.go:190]: actually minQuotas contains he data of  minResources. if we want to reduce similar fields:
1. may be we need a new field (we can named it minRequest) which calculated like minQuotas,  and then deprecated minResource field.
2. the overcommit, proportion plugin need to modify to use this new field.

Comment (k82cn) [reply pkg/apis/scheduling/types.go:190]: > may be we need a new field (we can named it minRequest) which calculated like minQuotas, and then deprecated minResource field.

So the differentiation is name? Why rename it?

Comment (merryzhou) [reply pkg/apis/scheduling/types.go:190]: the field meaning and he value of this filed are both changed

Comment (k82cn) [reply pkg/apis/scheduling/types.go:190]: let's sync up this in weekly meeting :)

**Comment by shinytang6** [comment]
Can we proceed this feature? We also need this feature a lot 🤕 
/cc @Thor-wl @k82cn @merryzhou

**Comment by Thor-wl** [comment]
Overall it's ok to me. But the related PR https://github.com/volcano-sh/volcano/pull/1345 has not been updated.

**Comment by k82cn** [comment]
> Can we proceed this feature? We also need this feature a lot 🤕 /cc @Thor-wl @k82cn @merryzhou

Can we book a meeting for that? I'm a little confused about the differentiation between minResource & minQuota :)

File: pkg/apis/scheduling/types.go (hunk 1 above)
Comment (Thor-wl) [reply pkg/apis/scheduling/types.go:190]: > 

Conclusion: reserve all fields from `minQuotas` to `minResource` @merryzhou

**Pull request closed without merging**